hal/armv7a/exceptions.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Exception handling
+ *
+ * Copyright 2017, 2018, 2021 Phoenix Systems
+ * Author: Pawel Pisarczyk, Jakub Sejdak, Aleksander Kaminski, Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+
+
+typedef struct _exc_context_t {
+	u32 savesp;
+
+	u32 dfsr;
+	u32 dfar;
+	u32 ifsr;
+	u32 ifar;
+
+	u32 psr;
+
+	u32 r0;
+	u32 r1;
+	u32 r2;
+	u32 r3;
+	u32 r4;
+	u32 r5;
+	u32 r6;
+	u32 r7;
+	u32 r8;
+	u32 r9;
+	u32 r10;
+
+	u32 fp;
+	u32 ip;
+	u32 sp;
+	u32 lr;
+
+	u32 pc;
+} exc_context_t;
+
+
+static const char digits[] = "0123456789abcdef";
+
+
+static int exceptions_i2s(char *prefix, char *s, unsigned int i, unsigned char b, char zero)
+{
+	char c;
+	unsigned int l, k, m;
+
+	m = hal_strlen(prefix);
+	hal_memcpy(s, prefix, m);
```

Comment:
```suggestion
	m = 0;
	while (*prefix) {
		s[m] = *prefix++;
		m++;
	}
```
will take less bytes

Reply:
Yeah, probably your proposition will save some bytes but you modify a `prefix` variable which should define as a `const` value (I will change it later ;).
In my opinion it should be left as it is, `hal_memcpy` and `hal_strlen` are standard functions and make reader knows exactly what is going on.